Minor Linting Corrections

euanwm opened the pull request:
Fixed some function comments and spelling mistakes.
Added additional test to validate a nil-Emitter scenario.

Raised a TODO in mint.go:21/22 as the context/mint.go passes an unhandled context.Err to the main function. Currently have this handled with a throwaway.

euanwm committed:
Added test workflow

euanwm committed:
Changed the wording slightly to be more readable

euanwm committed:
Merge branch 'btvoidx:main' into main

euanwm committed:
Merge branch 'btvoidx:main' into main

euanwm committed:
Comment tweaks

euanwm committed:
Comment tweaks #2

euanwm committed:
Throwaway handling

euanwm committed:
Simple test added

euanwm committed:
Verified cancel return value

btvoidx commented:
Thank you! Didn't realise I make that many typos.
Before I merge, could you elaborate on a TODO in [mint.go:21](https://github.com/btvoidx/mint/pull/3/files#diff-aea79064a692e530ab8e1b351c155227a4cc4cbc7d6c3c0496707a936822b0f1R21-R22)? `context.Background().Err()` is always nil, so I don't quite understand what you mean.

euanwm commented:
> Thank you! Didn't realise I make that many typos. Before I merge, could you elaborate on a TODO in [mint.go:21](https://github.com/btvoidx/mint/pull/3/files#diff-aea79064a692e530ab8e1b351c155227a4cc4cbc7d6c3c0496707a936822b0f1R21-R22)? `context.Background().Err()` is always nil, so I don't quite understand what you mean.

Regarding the TODO, do you want to pass the variable return by context/mint/Emit to the top level Emit function or keep it as it is?

After reading through the docs, the only reason for passing the Context.Err back would be if the function it's attached to hangs and it throws a DeadlineExceeded error or another edge-case situation. I'm probably wrong so I can easily drop the TODO if you're happy with the the rest.

btvoidx commented:
`context.Background().Err()` is always nil, so mint.go/Emit would always get back nil error from context/mint.go/Emit. I think it's better to ignore it explicitly with a throwaway, as various linters may yell at an unhandled error. Perhaps a comment explaining the reason should be added, but it's fine to just remove the todo.

euanwm committed:
Removed TODO

euanwm committed:
Defer in lopp caught

euanwm commented:
I've dropped out the TODO and I noticed that there was a defer in loop, wrapped it in an immediate anon function.

btvoidx committed:
fix plugin test to include call order

btvoidx committed:
Revert "Defer in lopp caught"

This reverts commit 571ea66c36cb64f3037fae66f90d646bf7991543.

btvoidx commented:
> I've dropped out the TODO and I noticed that there was a defer in loop, wrapped it in an immediate anon function.

`defer` call was working as intended. Wrapping a defer call to `after` in an immediate anon function breaks the call order, as all the calls to after-functions from plugins should happen in reverse order, just like `defer` does. I fixed bad test which didn't catch that and reverted your commit.

btvoidx commented:
Thank you for the corrections!

euanwm commented:
> > I've dropped out the TODO and I noticed that there was a defer in loop, wrapped it in an immediate anon function.
> 
> `defer` call was working as intended. Wrapping a defer call to `after` in an immediate anon function breaks the call order, as all the calls to after-functions from plugins should happen in reverse order, just like `defer` does. I fixed bad test which didn't catch that and reverted your commit.

I realised this was the intent after I had posted my comment. I'm a bit tired today.

No problem on the corrections 👍🏻
